include/llbuild/BuildSystem/BuildValue.h

```diff
@@ -113,14 +113,8 @@ class BuildValue {
   /// A hash value (used by some build value types).
   basic::CommandSignature signature;
 
-  union {
-    /// The file info for the rule output, for existing inputs, successful
-    /// commands with a single output, and directory contents.
-    FileInfo asOutputInfo;
-
-    /// The file info for successful commands with multiple outputs.
-    FileInfo* asOutputInfos;
-  } valueData = { {} };
+  /// The file info for successful commands with multiple outputs or a single output.
+  FileInfo* outputInfos;
```

Comment:
I like the simplification here removing the union, but perhaps it would be best in a separate PR?

Reply:
Hmm, a counterpoint for the simplification, I think the reason for the union is to avoid the allocation overhead on the many single output build values we might have floating around.